### PR TITLE
CAMS-390:  select lead case

### PR DESCRIPTION
**Pull request by fmaddenflx**
# Purpose

Allow user to select a Lead Case

After doing a round of validation testing with three users, we noticed that there were improvements needed to make consolidations more usable.

# Major Changes

Moved modal form out and placed in accordion for better UX.

Added ability to add Lead Case to accordion.

# Testing/Validation

90% coverage

### CODE REVIEW

File: user-interface/src/data-verification/ConsolidationCasesTable.test.tsx [hunk 1]
```diff
@@ -147,4 +170,66 @@ describe('test ConsolidationCasesTable component', () => {
     const alert = screen.queryByTestId('alert-container');
     expect(alert).not.toBeInTheDocument();
   });
+
+  // test selecting different "lead case" buttons and make sure
+  // - Button style and label is updated properly when different buttons are selected or clicked on and off
+  // - Button properly fires on click event from parent
```
Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationCasesTable.test.tsx:176]: Is this done?

File: user-interface/src/data-verification/ConsolidationCasesTable.test.tsx [hunk 2]
```diff
@@ -147,4 +170,66 @@ describe('test ConsolidationCasesTable component', () => {
...
+    });
+
+    let selectedLeadCaseIndex = 0;
+    leadCaseButton = screen.getByTestId(`button-assign-lead-${selectedLeadCaseIndex}`);
+    fireEvent.click(leadCaseButton);
+    expect(onMarkLead.mock.calls[0][0]).toEqual(props.cases[selectedLeadCaseIndex]);
+
+    leadCaseButtons.forEach((button) => {
+      if (leadCaseButton.id === button.id) {
+        console.log(button.className);
```
Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationCasesTable.test.tsx:203]:
```suggestion
```

File: user-interface/src/data-verification/ConsolidationCasesTable.test.tsx [hunk 3]
```diff
@@ -147,4 +170,66 @@ describe('test ConsolidationCasesTable component', () => {
...
+        expect(button).not.toHaveClass(UswdsButtonStyle.Outline);
+        expect(button).toHaveTextContent('Lead Case');
+      } else {
+        console.log(button.className);
```
Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationCasesTable.test.tsx:207]:
```suggestion
```

File: user-interface/src/data-verification/ConsolidationOrderAccordion.test.tsx [hunk 4]
```diff
@@ -171,51 +259,206 @@ describe('ConsolidationOrderAccordion tests', () => {
...
+      expect(approveButton).toBeEnabled();
+      expect(rejectButton).toBeEnabled();
+    });
+
+    ///////////// Now testing "case not listed" ///////////////
+
+    const markAsLeadButton = screen.getByTestId('button-assign-lead-0');
+    expect(markAsLeadButton).not.toHaveClass('usa-button--outline');
+
+    // selecting "Lead Case Not Listed" should unselect the "MarkAsLead" button
```
Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationOrderAccordion.test.tsx:324]: Do we need the comments here? In the second case, maybe we could extract a function that makes it clear what we're testing for.

File: user-interface/src/data-verification/ConsolidationOrderAccordion.tsx [hunk 5]
```diff
@@ -43,21 +90,132 @@ export interface ConsolidationOrderAccordionProps {
 
 export function ConsolidationOrderAccordion(props: ConsolidationOrderAccordionProps) {
   const { hidden, statusType, orderType, officesList, expandedId } = props;
-  const caseTable = useRef<OrderTableImperative>(null);
 
-  const confirmationModalRef = useRef<ConfirmationModalImperative>(null);
+  //////////////////// REFS ///////////////////
```
Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationOrderAccordion.tsx:94]: I'm not necessarily opposed to this. But could we make them a bit more uniform, like using a block comment or something?
```suggestion
  /***** REFS *****/
```

Comment (jamesobrooks) [reply user-interface/src/data-verification/ConsolidationOrderAccordion.tsx:94]: Although I do wonder if there's a way we could instead enforce grouping these things via linting.

File: user-interface/src/data-verification/ConsolidationOrderModal.test.tsx [hunk 6]
```diff
@@ -27,17 +21,17 @@ describe('ConsolidationOrderModalComponent', () => {
...
 
-  function enterCaseNumberInModal(caseIdInput: Element | null | undefined, value: string) {
-    if (!caseIdInput) throw Error();
+  //function enterCaseNumberInModal(caseIdInput: Element | null | undefined, value: string) {
+  //  if (!caseIdInput) throw Error();
 
-    fireEvent.change(caseIdInput!, { target: { value } });
-  }
+  //  fireEvent.change(caseIdInput!, { target: { value } });
+  //}
```
Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationOrderModal.test.tsx:34]: I believe we can remove this now.

File: user-interface/src/data-verification/ConsolidationOrderModal.test.tsx [hunk 7]
```diff
@@ -322,193 +203,194 @@ describe('ConsolidationOrderModalComponent', () => {
...
+  //  await waitFor(() => {
+  //    expect(caseNumberInput).toHaveValue(leadCaseNumber);
+  //  });
+
+  //  const alertElement = screen.queryByTestId('alert-message');
+  //  expect(alertElement).toHaveTextContent(
+  //    'Cannot verify lead case is not part of another consolidation. Test error.',
+  //  );
+  //  expect(continueButton).not.toBeEnabled();
+  //});
```
Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationOrderModal.test.tsx:395]:
```suggestion
```

Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationOrderModal.test.tsx:395]: Let's remove these.

File: user-interface/src/data-verification/ConsolidationOrderModal.tsx [hunk 8]
```diff
@@ -69,32 +51,16 @@ export type ConfirmationModalImperative = ModalRefType & {
...
-  return divisionCodeSet.size === 1 ? Array.from<string>(divisionCodeSet)[0] : undefined;
-}
+//export async function getCaseAssignments(caseId: string) {
+//  return useGenericApi().get<Array<CaseAssignment>>(`/case-assignments/${caseId}`);
+//}
+//
+//export async function getCaseAssociations(caseId: string) {
+//  return useGenericApi().get<Array<Consolidation>>(`/cases/${caseId}/associated`);
+//}
+//
```
Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationOrderModal.tsx:61]: Let's remove this.

File: user-interface/src/data-verification/ConsolidationOrderModal.tsx [hunk 9]
```diff
@@ -195,56 +140,29 @@ function ConsolidationOrderModalComponent(
...
-
-      case 'rejected':
-        modalRef.current?.buttons?.current?.disableSubmitButton(false);
-        setOptions({
-          status: options.status,
-          heading: 'Reject Case Consolidation?',
-        });
-        break;
+    if (options.status === 'approved' && options.leadCase && options.consolidationType) {
+      // modalRef.current?.buttons?.current?.disableSubmitButton(false);
```
Comment (jamesobrooks) [inline user-interface/src/data-verification/ConsolidationOrderModal.tsx:144]: I think we can remove this too.

File: user-interface/src/data-verification/ConsolidationCasesTable.test.tsx (hunk 2 above)
Comment (fmaddenflx) [reply user-interface/src/data-verification/ConsolidationCasesTable.test.tsx:205]: resolved.

File: user-interface/src/data-verification/ConsolidationCasesTable.test.tsx (hunk 3 above)
Comment (fmaddenflx) [reply user-interface/src/data-verification/ConsolidationCasesTable.test.tsx:209]: resolved.

File: user-interface/src/data-verification/ConsolidationOrderAccordion.tsx (hunk 5 above)
Comment (fmaddenflx) [reply user-interface/src/data-verification/ConsolidationOrderAccordion.tsx:94]: I was trying to make them stick out when scrolling quickly through the code.  I can change to a block comment though, that's fine.

File: user-interface/src/data-verification/ConsolidationOrderAccordion.test.tsx (hunk 4 above)
Comment (fmaddenflx) [reply user-interface/src/data-verification/ConsolidationOrderAccordion.test.tsx]: extracting a function at this point would be quite messy, parameter-wise.

Comment (fmaddenflx) [reply user-interface/src/data-verification/ConsolidationOrderAccordion.test.tsx]: I'll split it into 2 tests.

File: user-interface/src/data-verification/ConsolidationOrderAccordion.tsx (hunk 5 above)
Comment (fmaddenflx) [reply user-interface/src/data-verification/ConsolidationOrderAccordion.tsx:94]: Actually I do have an argument against this:  Using block comments makes things difficult or annoying when you actually want to use block comments to comment something out for a test or whatever, and you don't realize that there is another block comment within it.  Then you have to spend a few minutes figuring out why everything is broken.

I'd rather reserve block comments for rare occasions when you are purposefully trying to block out a large portion of the code.

Comment (fmaddenflx) [reply user-interface/src/data-verification/ConsolidationOrderAccordion.tsx:94]: We could do something like //================== REFS ==================

**Review by btposey — APPROVED** [review]

**Pull request merged**